tokio-console/src/view/mod.rs

```diff
@@ -168,32 +177,56 @@ impl View {
         update_kind
     }
 
+    fn handl_help_popup(&mut self, event: crossterm::event::Event) {
+        if input::is_help_toggle(&event) {
+            // TODO: Pause state if we are about to show the help toggle
+            self.show_help_toggle = !self.show_help_toggle;
+        }
+        if self.show_help_toggle && !input::is_help_toggle(&event) {
+            // We have some other key pressed; clear the help popup.
+            self.show_help_toggle = !self.show_help_toggle;
+        }
+    }
+
     pub(crate) fn render<B: tui::backend::Backend>(
         &mut self,
         frame: &mut tui::terminal::Frame<B>,
         area: layout::Rect,
         state: &mut State,
     ) {
+        let mut help_content;
         match self.state {
             ViewState::TasksList => {
                 self.tasks_list.render(&self.styles, frame, area, state, ());
+                help_content = HelpView::new(TableListState::<TasksTable>::render_help_content(
+                    &self.styles,
+                ));
             }
             ViewState::ResourcesList => {
                 self.resources_list
                     .render(&self.styles, frame, area, state, ());
+                help_content = HelpView::new(
+                    TableListState::<ResourcesTable>::render_help_content(&self.styles),
+                );
             }
             ViewState::TaskInstance(ref mut view) => {
                 let now = state
                     .last_updated_at()
                     .expect("task view implies we've received an update");
                 view.render(&self.styles, frame, area, now);
+                help_content = HelpView::new(TaskView::render_help_content(&self.styles));
             }
             ViewState::ResourceInstance(ref mut view) => {
                 view.render(&self.styles, frame, area, state);
+                help_content = HelpView::new(ResourceView::render_help_content(&self.styles));
```

Comment:
I notice that this will construct the help content for each view *every* time we render a screen, even if we're not showing the help window. That seems a little bit inefficient, since we have to allocate a big `vec` and a bunch of strings.

I think it might be more efficient to add a second `match self.state` inside of the `if self.show_help_toggle` box, and only build the help text if we are actually going to display a help box. That might make the method a bit longer, but I think it's worth doing to avoid having to format the help text when we're not going to display it...